applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -639,7 +639,7 @@ void UPwLysmerAbsorbingCondition<3, 4>::CalculateRotationMatrix( BoundedMatrix<d
     //Quadrilateral_3d_4
     array_1d<double, 3> p_mid_0;
     array_1d<double, 3> p_mid_1;
-    const auto& r_p_2 = array_1d<double, 3>(rGeom.GetPoint(2));
+    const auto& r_p_2 = rGeom.GetPoint(2);
```

Comment:
```suggestion
    const array_1d<double, 3>& r_p_2 = rGeom.GetPoint(2);
```

I recommend not to use `auto` with `array_1d<double, 3>`, as in the past GCC had issues deriving the correct types, leading to very strange behaviors

Reply:
Hi @philbucher, thanks for pointing out this issue, I was not aware of that. Do you recall which version of GCC that was? Could it be that the issue has been resolved by now, since I have no failing builds (including GCC builds)?

In general, I'm not in favor of explicitly repeating type names, since that makes the code less flexible with respect to future changes and extensions. I will discuss the point that you've raised with one of my colleagues tomorrow to see what is the best way to proceed. Thanks.

Reply:
See #7414

I think it was with GCC8/9/10, but do not remember

I think [this](https://github.com/KratosMultiphysics/Kratos/issues/7414#issuecomment-704121996) is a good guideline, so in your case it would most probably be ok

Reply:
I have discussed your comment with a colleague and I feel that we need to have a deeper understanding here. To us it seems that the problem that you had raised in the past (#7414) may be related to lazy evaluation of the expression. We have the impression that by being explicit about the type the evaluation may be triggered (in particular when a type conversion is required like here). Again, we need to dig deeper first to gain a better understanding. For now I will go with your suggestion, but I may come back to it later. Thanks anyway for your comments.